app/lib/features/chat_ng/providers/chat_room_messages_provider.dart

```diff
@@ -194,9 +192,49 @@ final mentionSuggestionsProvider =
       };
     });
 
-final repliedToMsgProvider = AsyncNotifierProvider.autoDispose
-    .family<RepliedToMessageNotifier, RepliedToMsgState, RoomMsgId>(() {
-      return RepliedToMessageNotifier();
+/// Give the room and the message id that contains a reply, this returns
+/// the event item that is being replied to.
+final repliedToMsgProvider = FutureProvider.autoDispose
+    .family<TimelineEventItem, RoomMsgId>((ref, item) async {
+      TimelineItem? msg = ref.watch(chatRoomMessageProvider(item));
+      TimelineEventItem? repliedToItem;
+
+      if (msg == null) {
+        throw 'Event ${item.uniqueId} not found';
+      }
+      final inReplyToId = msg.eventItem()?.inReplyToId();
+      if (inReplyToId != null) {
+        repliedToItem =
+            ref
+                .watch(
+                  chatRoomMessageProvider((
+                    roomId: item.roomId,
+                    uniqueId: inReplyToId,
+                  )),
+                )
+                ?.eventItem();
+        if (repliedToItem != null) {
+          return repliedToItem;
+        }
+      }
+
+      // not found in the history, let's try to read directly
+
+      repliedToItem = msg.eventItem()?.inReplyToEvent();
+      if (repliedToItem == null) {
+        // failed direclty, trying to fetch from remote then
+        final timeline =
+            ref.read(chatMessagesStateProvider(item.roomId).notifier).timeline;
+        await timeline.fetchDetailsForEvent(item.uniqueId);
+        repliedToItem = (await timeline.getMessage(item.uniqueId)).eventItem();
```

Comment:
if that isn't there either, we ask the timeline to fetch the information for us via this new API